pool/pool.go

```diff
@@ -223,12 +228,57 @@ func (p *Pool) StoreTx(ctx context.Context, tx types.Transaction, ip string, isW
 		}
 	}
 
+	gasPrices, err := p.GetGasPrices(ctx)
+	if err != nil {
+		log.Error("failed to load gas prices")
+	}
+	err = p.ValidateEffectiveGasPrice(ctx, tx, preExecutionResponse.txResponse.GasUsed, gasPrices)
```

Comment:
I think this function must be calle `ValidateBreakEvenGasPrice` as really this function is checking the breakEvenGasPrice, the EffectiveGasPrice is a different calculation/value that takes also into account the ratio between tx.GasPrice and L2GasPrice